### PR TITLE
OLS-650: Ability to add additional CA certs

### PR DESCRIPTION
**Pull request by onmete**
## Description

Ability to add additional CA certs

## Type of change

- [x] New feature

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,29 +160,40 @@ OpenShift LightSpeed (OLS) is an AI powered assistant that runs on OpenShift and
 
 9.  (Optional) Configure conversation cache
    Conversation cache can be stored in memory (it's content will be lost after shutdown) or in PostgreSQL database. It is possible to specify storage type in `olsconfig.yaml` configuration file.
-   1. Cache stored in memory:
+
```
Comment (onmete) [inline README.md:163]: This block had a wrong indentation.

File: runner.py [hunk 2]
```diff
@@ -47,6 +49,30 @@ def load_index():
...
 
 
+# NOTE: Be aware this is irreversible step, as it modifies the certifi
+# store. To restore original certs, the certifi lib needs to be
+# reinstalled (or extra cert manually removed from the file).
+def add_ca_to_certifi(cert_path: str, certifi_cert_location=certifi.where()) -> None:
+    """Add a certificate to the certifi store."""
+    print(f"Certifi store location: '{certifi_cert_location}'")
+    print(f"Adding certificate '{cert_path}' to certifi store")
+    with open(cert_path, "rb") as certificate_file:
```
Comment (tisnik) [inline runner.py:59]: are not these files regular text files?

File: tests/unit/extra_certs/test_extra_certs.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        with open(cert_in_certitifi_store_path, "rb") as cert_file:
+            cert_store.write(cert_file.read())
+    return cert_store_path.strpath
+
+
+def load_certs(cert_path):
+    """Load a certificates from a file."""
+    with open(cert_path, "rb") as cert_file:
+        cert_data = cert_file.read()
+    certs = x509.load_pem_x509_certificates(cert_data)
```
Comment (tisnik) [inline tests/unit/extra_certs/test_extra_certs.py:26]: simple return

**Review by tisnik — COMMENTED** [review]
just one question and one nitpick, but look ok in overall!

File: runner.py (hunk 2 above)
Comment (onmete) [reply runner.py:59]: Yes, but the cryptography functions work with bytes.

**Review by tisnik — COMMENTED** [review]
it looks ok!

**Pull request merged**